0001-typedb/02-dev/05-read.md

```diff
@@ -0,0 +1,395 @@
+
+TypeDB supports the following types of aggregation:
+
+- count
+- sum
+- max
+- mean
+- median
+
+To perform aggregation in TypeDB, we first write a [match clause](03-match.md) to describe the set of data, then follow 
+that by get to retrieve a distinct set of answers based on the specified variables, and lastly an aggregate 
+function to perform on the variable of interest.
+
+<div class="note">
+[Note]
+Aggregation uses data returned by the query to perform the calculation. 
+
+For example:
+
+<!-- test-ignore -->
+```typeql 
+match $f isa file, has size-kb $s; get $s;
+```
+
+If you query the example above, only attribute `$s` will be returned with its value. But all the duplicated values of 
+`$s` will be omitted. Because what we are querying is all attribute instances. And when two different entities have 
+the same attribute (with the same value) it's still only one attribute.
+
+So it’s just like to query only unique values. For example, if there are three files with the same value of `size-kb` 
+attribute = 10, we will get only one of this value.
+
+if we want to prevent that we should return not only attributes, but also the entities that own the attributes in the 
+`get` clause. For example:
+
+<!-- test-ignore -->
+```typeql 
+match $f isa file, has size-kb $s; get $f, $s;
+```
+
+This will return pairs of `$f` and `$s`, including all duplicated values of `$s` (every `file` instance with its 
+attribute of `size-kb`. Including all of the same values owned by different files. For example, all three values of 10 
+will be returned this way).
+</div>
+
+#### Count
+
+Use the count keyword to get the number of the specified matched variable.
+
+<!-- test-ignore -->
+```typeql 
+match
+  $o isa object, has path $fp;
+get $o, $fp; count;
+```
+
+<div class="note">
+[Note]
+The `count` function is applied to every result returned. If more than one variable mentioned in get, then `count` 
+will show the number of unique combinations of results. This is also the case when no `get` clause is added, which 
+actually means that all matched variables are included.
+</div>
+
+<!-- test-ignore -->
+```typeql 
+match
+  $pe ($x, $y) isa permission;
+  $x isa person, has full-name $x-n;
+  $y (accessed-object: $o, valid-action: $act) isa access;
+  $act has action-name $act-n;
+  $o has path $o-fp;
+get $x-n, $act-n, $o-fp; group $o-fp; count;
+```
+
+This query returns the total count of `person` instances that have `full-name` as well as any `access` to an `object` 
+with `path` and with a `valid action` for every group (grouped by the `path` of the `object`).
+
+<div class="note">
+[Note]
+The `group` clause should go before the aggregation function.
+</div>
+
+#### Sum
+
+Use the `sum` keyword to get the sum of the specified `long` or `double` values of matched variable.
+
+<!-- test-ignore -->
+```typeql 
+match
+  $f isa file, has size-kb $s;
+get $f, $s; 
+sum $s;
+```
```

Comment:
Do you think that would be enough? https://github.com/vaticle/docs/pull/676/commits/c2b74d30e03a945ed34ab43365c0dfa8917c3b56

Reply:
It's up to you. I would've written even more, since this behaviour is so different to SQL and so subtle. As I noted, even though our current documents are very, very misleading (wrong, really), how many people have actually noticed this detail?

Reply:
I think it is enough, because in our new docs we actually have a better explanation of matching patterns that cause this behavior.